test/stop/residual_norm_kernels.cpp

```diff
@@ -110,6 +110,45 @@ TYPED_TEST(ResidualNorm, CanIgorneResidualNorm)
                  gko::NotSupported);
 }
 
+TYPED_TEST(ResidualNorm, CheckIfResZeroConverges)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using NormVector = typename TestFixture::NormVector;
+    auto initial_res = gko::initialize<Mtx>({10.0}, this->exec);
+    std::shared_ptr<gko::LinOp> rhs = gko::initialize<Mtx>({10.0}, this->exec);
+    auto criterion =
+        this->factory->generate(nullptr, rhs, nullptr, initial_res.get());
+    auto rel_criterion =
+        this->rel_factory->generate(nullptr, rhs, nullptr, initial_res.get());
```

Comment:
you need to set the reduction_factor to zero such that it indeed uses `=`.
For all cases with the original issue corcern,
1. absolute: reduction_factor = 0,
2. initial_resnorm: initial_res = 0 
3. rhs_norm: rhs = 0

create the new absolute, initial_resnorm, rhs_norm here not from the object in TestFixture.
You may need to add `using ValueType = T;` in class